### PR TITLE
remove strictPropertyInitialization in components

### PR DESCRIPTION
**Pull request by m4thieulavoie**
# Pull Request

## 📖 Description

This PR removes the `strictPropertyInitialization` rule from `fast-component`'s `tsconfig` file.

### 🎫 Issues

https://github.com/microsoft/fast/issues/5339

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-components/src/avatar/index.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class Avatar extends FoundationAvatar {
      * HTML Attribute: src
      */
     @attr({ attribute: "src" })
-    public imgSrc: string;
+    public imgSrc = "";
```
Comment (robarbms) [inline packages/web-components/fast-components/src/avatar/index.ts:23]: Any reason to not keep the typings, especially initialized, they won't be null or undefined?
```suggestion
    public imgSrc: string = "";
```

**Review by robarbms — APPROVED** [review]

File: packages/web-components/fast-components/src/avatar/index.ts (hunk 1 above)
Comment (m4thieulavoie) [reply packages/web-components/fast-components/src/avatar/index.ts:23]: I was unsure on what y'all prefer on this one (I've been at companies that prefer either one). Happy to put it back 😄

Comment (m4thieulavoie) [inline packages/web-components/fast-components/src/avatar/index.ts]:
```suggestion
    public imgSrc: string = "";
```

File: packages/web-components/fast-components/src/avatar/index.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class Avatar extends FoundationAvatar {
      * @remarks
      * HTML Attribute: alt
      */
-    @attr public alt: string;
+    @attr public alt = "";
```
Comment (m4thieulavoie) [inline packages/web-components/fast-components/src/avatar/index.ts:32]:
```suggestion
    @attr public alt: string = "";
```

File: packages/web-components/fast-components/src/disclosure/index.ts [hunk 3]
```diff
@@ -17,11 +17,11 @@ export class Disclosure extends FoundationDisclosure {
     /**
      * Disclosure default height
      */
-    private height: number;
+    private height = 0;
```
Comment (m4thieulavoie) [inline packages/web-components/fast-components/src/disclosure/index.ts:20]:
```suggestion
    private height: number = 0;
```

File: packages/web-components/fast-components/src/disclosure/index.ts [hunk 4]
```diff
@@ -17,11 +17,11 @@ export class Disclosure extends FoundationDisclosure {
     /**
      * Disclosure default height
      */
-    private height: number;
+    private height: number = 0;
     /**
      * Disclosure height after it's expanded
      */
-    private totalHeight: number;
+    private totalHeight = 0;
```
Comment (m4thieulavoie) [inline packages/web-components/fast-components/src/disclosure/index.ts:24]:
```suggestion
    private totalHeight: number = 0;
```

File: packages/web-components/fast-components/src/design-system-provider/index.ts [hunk 5]
```diff
@@ -188,7 +188,7 @@ export class DesignSystemProvider extends FoundationElement {
         converter: swatchConverter,
     })
     @designToken(fillColor)
-    public fillColor: Swatch;
+    public fillColor: Swatch = fillColor.getValueFor(this);
```
Comment (nicholasrice) [inline packages/web-components/fast-components/src/design-system-provider/index.ts:191]: Doing this is going to cause a lot of un-necessary work for each design-system-provider instance so I don't think we should do this. 

I'd propose updating the property types to allow for undefined, thus not requiring initialization (at least I think that is how it works).  If that *doesn't* work, I'd propose disabling that rule for this file.

Comment (m4thieulavoie) [reply packages/web-components/fast-components/src/design-system-provider/index.ts]: That should be done 👍

**Comment by m4thieulavoie** [comment]
@chrisdholt can you have a look at this one? :pray:

File: packages/web-components/fast-components/src/avatar/index.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ export class Avatar extends FoundationAvatar {
      * HTML Attribute: src
      */
     @attr({ attribute: "src" })
-    public imgSrc: string;
+    public imgSrc: string = "";
```
Comment (chrisdholt) [inline packages/web-components/fast-components/src/avatar/index.ts:23]: Should this perhaps be null or undefined?

**Review by chrisdholt — APPROVED** [review]
Would be great to get @nicholasrice or @EisenbergEffect’s eyes in this as well. Thanks @m4thieulavoie!

File: packages/web-components/fast-components/src/avatar/index.ts (hunk 6 above)
Comment (m4thieulavoie) [reply packages/web-components/fast-components/src/avatar/index.ts:23]: Yeah good call, it should be consistent with the remaining of the PR :+1:

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-components/src/anchor/index.ts [hunk 7]
```diff
@@ -25,7 +25,7 @@ export class Anchor extends FoundationAnchor {
      * HTML Attribute: appearance
      */
     @attr
-    public appearance: AnchorAppearance;
+    public appearance: AnchorAppearance = "neutral";
```
Comment (EisenbergEffect) [inline packages/web-components/fast-components/src/anchor/index.ts:28]: For some reason I thought we couldn't do this and we had to check in the connectedCallback. Maybe it was an old bug that was fixed...I though there was some sort of timing issue with upgrades or binding. Something like that. @nicholasrice Does any of this ring a bell for you?

Comment (nicholasrice) [reply packages/web-components/fast-components/src/anchor/index.ts:28]: hmmm maybe? I didn't see the classList mutation, that may be why it was done this way. It is probably best to keep the assignment in the property initialization and just move the class to the template.

**Review by EisenbergEffect — APPROVED** [review]

**Pull request merged**